crates/holochain/tests/tests/countersigning.rs

```diff
@@ -289,6 +289,115 @@ async fn retry_countersigning_commit_on_missing_deps() {
     wait_for_completion(bob_rx, preflight_request.app_entry_hash).await;
 }
 
+// Checks that when there are multiple authorities returning signature bundles and multiple sessions
+// happening between a pair of agents, the extra signature bundles beyond the first one received
+// are correctly handled. I.e. receiving further signature bundles after a new session has started
+// will not impact the new session.
+// This is trying to check a race condition, so it's probably expected that the test won't always
+// correctly test the problem. However, it is a scenario we've seen fail in Wind Tunnel, so it's
+// worth trying to test here. Seeing flakes from this test is likely a signal that something has
+// regressed.
+#[tokio::test(flavor = "multi_thread")]
+async fn signature_bundle_noise() {
```

Comment:
I'm happy to report that this test reliably fails on current `develop`. I.e. transplanting just the test and not the fix. I wasn't 100% confident in my fix until I saw that :)